### DISCUSSION TITLE
How can I suppress warning messages?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1603] [Q&A] [answered]
运行xrepo env shell时，如果出现warning，则会导致prompt的探测出问题。现在powershell通过运行`xmake l private.xrepo.action.env.info prompt`获取prompt，但这个命令会输出warning信息，导致prompt包含warning，输出错误结果。

问题：如何在xmake中抑制warning message，或者将warning视为error，或者把warning输出到stderr而不是stdout？

**Comment by waruqi** [comment, marked as answer]
什么警告，压不了，要么试试 --quiet

**Reply by xq114** [reply]
warning: set_toolchain(msvc, nil) is deprecated, please uses set_toolset(msvc, nil) instead of it

类似这种。运行`xmake l private.xrepo.actions.env.info prompt`输出的时候会输出
```
[xmake-test]
warning: set_toolchain(msvc, nil) is deprecated, please uses set_toolset(msvc, nil) instead of it
```
但想要的输出是
```
[xmake-test]
```
加`--quiet`就什么输出都没有了

**Comment by waruqi** [comment]
--quiet 下，走 io.write 原始输出接口，绕过 quiet

```lua
    if key == "prompt" then
        assert(os.isfile(os.projectfile()), "xmake.lua not found!")
        io.write("[" .. path.filename(os.projectdir()) .. "]")
```
